File: challenge_1/challenge1.js
const readline = require('readline');

const terminal = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

let inputLines = [];
let maxInt = 0;
let radius = 0;
terminal.on('line', (line) => {
    inputLines.push(line.trim());
    if (inputLines.length === 1) {
        maxInt =  parseInt(inputLines[0], 10);
        radius = maxInt / 2
    }
    if (inputLines.length === maxInt + 1) { // accounting for the first index being used by maxInt
        let array = inputLines.slice(1, maxInt + 1);
        let result = 0;
        for (let i = 0; i < radius; i++) {
            if (array[i] === array[i + radius]) {
                result += 2;
            }
        }

        console.log(result);
        terminal.close();
    }
});

File: challenge_1/challenge1.dart
import 'dart:io';
void main() {
  int maxInt = int.parse(stdin.readLineSync() ?? '');
  int radius = maxInt ~/ 2;
  List<String> array = [];
  for (int i = 0; i < maxInt; i++) {
    array.add(stdin.readLineSync() ?? '');
  }

  int result = 0;
  for (int i = 0; i < radius; i++) {
    if (array[i] == array[i+radius]) {
      result += 2;
    }
  }
  print(result);
}